CMakeLists.txt

```diff
@@ -0,0 +1,571 @@
+# # CMake project for fdk-aac
+#
+# ## Basic usage
+#
+# 1. Download and install CMake >= 3.10 (CMake >= 3.15 is recommended)
+# 2. Clone fdk-aac repository
+# 3. In fdk-aac directory create and go to `build` subdirectory:
+#
+#    mkdir build
+#    cd build
+#
+# 4. Run cmake to configure project with desired build type (Release|Debug):
+#
+#    cmake .. -DCMAKE_BUILD_TYPE=Release
+#
+#    Note CMake configuration without CMAKE_BUILD_TYPE option will not set build
+#    type to some default and it is most likely is not what you want.
+#
+#    If you want to build DLL (default is static library), add BUILD_SHARED_LIBS=ON option:
+#
+#    #    cmake .. -DBUILD_SHARED_LIBS=ON
+#
+#    You can combine options as well:
+#
+#    cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON
+#
+# 5. Run cmake to build project:
+#
+#    cmake --build . -c Release
+
+cmake_minimum_required(VERSION 3.10)
+
+# Policies
+
+## Enables CMAKE_MSVC_RUNTIME_LIBRARY option support for CMake >= 3.15
+## if you want to use a MSVC multi-threaded statically-linked runtime library
+## If you enable it, CMake will build fdk-acc.dll without external dependencies.
+##
+## Example usage:
+##
+## cmake .. -DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>
+cmake_policy(SET CMP0091 NEW)
+
+project(fdk-aac VERSION 2.0.1)
+
+# Includes
+
+include(CheckLibraryExists)
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(CPack)
```

Comment:
Afaik the `CPack` include here is mostly unused so far - I think I'd omit it for now then (and package source releases with autotools still).

Reply:
Why not? It could be used to package not only sources, but also DEBs, PRMs and many more. For Windows you can create ZIP with DLL, headers and all other installed files by one command, very useful when you create GitHub release.

Reply:
For now, the CPack packaging is unconfigured so it gives the packages version 0.1.1 or something like that.

Also I'm not planning on binary releases (the code has a tricky license so I avoid redistributing binaries at all).

But if CPack is properly configured I don't mind keeping it in the CMakeLists.txt for others to use - but as it is right now it's missing some configuration IMO.

Reply:
> For now, the CPack packaging is unconfigured so it gives the packages version 0.1.1 or something like that.

Known issue, i will fix that.